hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -697,7 +738,13 @@ public HoodieCDCSupplementalLoggingMode cdcSupplementalLoggingMode() {
   }
 
   public String getKeyGeneratorClassName() {
-    return getString(KEY_GENERATOR_CLASS_NAME);
+    String keyGeneratorClassName = null;
+    if (contains(KEY_GENERATOR_TYPE)) {
+      keyGeneratorClassName = KeyGeneratorType.valueOf(getString(KEY_GENERATOR_TYPE)).getClassName();
+    } else if (contains(KEY_GENERATOR_CLASS_NAME)) {
+      keyGeneratorClassName = getString(KEY_GENERATOR_CLASS_NAME);
+    }
```

Comment:
Move this into `KeyGeneratorType`

Reply:
please check my comment about not keeping this logic in enum at https://github.com/apache/hudi/pull/9590/files#r1321299763